pkg/plugin/datasource.go

```diff
@@ -89,30 +89,35 @@ func newDatasourceInstance(ctx context.Context, settings backend.DataSourceInsta
 		return nil, fmt.Errorf("failed to parse datasource url: %w", err)
 	}
 
-	var dsOpts struct {
-		QueryParams string `json:"customQueryParameters,omitempty"`
-		VMUIURL     string `json:"vmuiUrl,omitempty"`
-	}
-	if err := json.Unmarshal(settings.JSONData, &dsOpts); err != nil {
+	var dstSettings DataSourceInstanceSettings
+	if err := json.Unmarshal(settings.JSONData, &dstSettings); err != nil {
 		return nil, fmt.Errorf("failed to parse datasource settings: %w", err)
 	}
-	queryParams, err := url.ParseQuery(dsOpts.QueryParams)
+	queryParams, err := url.ParseQuery(dstSettings.QueryParams)
```

Comment:
Thanks for doing this! But I think it deserves a separate PR. Because the current commit message and changelog line do not cover all the changes made.
I suggest doing the settings PR first
and $__rate_interval change afterwards

Reply:
Moved changes related datasource settings to the new PR
https://github.com/VictoriaMetrics/victoriametrics-datasource/pull/349/files